### PR TITLE
Add "how to render a component to a string" to FAQ

### PR DESCRIPTION
**Pull request by Spone**
### Summary

It's not obvious that you cannot use `render` or `render_to_string` to render a ViewComponent from inside a controller action.

### Other Information

The `#render_in` method is private, but I can't find any other way to achieve this. I think it's a good idea to mention this in the FAQ (especially to help people debug `AbstractController::DoubleRenderError`), but maybe add a disclaimer? Or consider making the method public?

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -66,7 +70,7 @@ title: Changelog
 
 * Add test case for conflict with internal `@variant` variable.
 
-   *David Backeus*
+    *David Backeus*
```
Comment (vpereira) [inline docs/CHANGELOG.md:73]: looks like this change is irrelevant to the PR?

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -130,7 +134,7 @@ title: Changelog
 
 * Ensure consistent indentation with Rubocop.
 
-    *Joel Hawksley
+    *Joel Hawksley*
```
Comment (vpereira) [inline docs/CHANGELOG.md:137]: looks like this change is irrelevant to the PR?

File: docs/CHANGELOG.md (hunk 1 above)
Comment (Spone) [reply docs/CHANGELOG.md:73]: Yes, I was just fixing the changelog indentation at the same time

Comment (vpereira) [reply docs/CHANGELOG.md:73]: Ok, normally in that case we should add these "unrelated" changes to a separate commit, like "Fix indentation"

**Comment by Spone** [comment]
Here is a related comment: https://github.com/github/view_component/issues/201#issuecomment-582413280 where @joelhawksley explains that:

> `render_in` is the public interface for components - it's definitely the right way in!

File: docs/CHANGELOG.md (hunk 2 above)
Comment (joelhawksley) [reply docs/CHANGELOG.md:137]: I'm fine with landing this formatting fix :)

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,10 @@ title: Changelog
 * Allow query parameters in `with_request_url` test helper.
 
     *Javi Martín*
+    
+* Add "how to render a component to a string" to FAQ
```
Comment (joelhawksley) [inline docs/CHANGELOG.md:14]:
```suggestion
* Add "how to render a component to a string" to FAQ.
```

File: docs/CHANGELOG.md (hunk 1 above)
Comment (joelhawksley) [reply docs/CHANGELOG.md:73]: We squash commits on merge, so I'm not too worried about atomic commits unless it's a big PR that might be best reviewed step-wise. Even then, we'd probably want to multiple PRs ❤️

File: docs/frequently-asked-questions.md [hunk 4]
```diff
@@ -13,6 +13,24 @@ Yes. ViewComponent is tested against ERB, Haml, and Slim, but it should support
...
+class PagesController < ApplicationController
+  def index
+    # Does not work: triggers a `AbstractController::DoubleRenderError`
+    # @reusable_icon = render IconComponent.new('close')
+    
+    # Does not work: renders the whole index view as a string
+    # @reusable_icon = render_to_string IconComponent.new('close')
+  
+    # Works: renders the component as a string
+    @reusable_icon = IconComponent.new('close').render_in(view_context)
```
Comment (joelhawksley) [inline docs/frequently-asked-questions.md:30]: Woah, I didn't even think about this use case. That is tricky!

**Review by joelhawksley — APPROVED** [review]

File: docs/CHANGELOG.md [hunk 5]
```diff
@@ -14,6 +14,10 @@ title: Changelog
 * Allow query parameters in `with_request_url` test helper.
 
     *Javi Martín*
+
```
Comment (joelhawksley) [inline docs/CHANGELOG.md:17]:
```suggestion

```

**Comment by Spone** [comment]
@joelhawksley what do you think about documenting `#render_in` in https://viewcomponent.org/api.html#instance-methods?

**Comment by joelhawksley** [comment]
@Spone sure thing! The API docs need some attention for sure ❤️

**Pull request merged**